teamcity/resource_build_config.go

```diff
@@ -826,11 +826,19 @@ func expandBuildFeature(raw interface{}) (api.BuildFeature, error) {
 func expandBuildSteps(list interface{}) ([]api.Step, error) {
 	out := make([]api.Step, 0)
 	in := list.([]interface{})
+	names := make(map[string]bool)
```

Comment:
FYI you can also use the empty struct as the map value type (i.e. `map[string]struct{}`), which is technically better memory-wise, and makes this more strictly a "set" type. Keeping it as a `bool` also has its advantages -- see below -- up to you which you prefer here!

Reply:
New trick learned!